keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -287,6 +278,23 @@ open class SqlResourceRepository(
     }
   }
 
+  override fun <T : ResourceEvent> lastEventByType(id: String, eventType: Class<T>): T? {
+    return sqlRetry.withRetry(READ) {
+      jooq
+        .select(EVENT.JSON)
+        .from(EVENT, RESOURCE)
+        .where(EVENT.SCOPE.eq(Scope.RESOURCE.name))
+        .and(RESOURCE.ID.eq(id))
+        .and(RESOURCE.UID.eq(EVENT.UID))
+        .and("event.json->'\$.type' = '${ResourceDeltaResolved::class.simpleName}'")
```

Comment:
Nice to see the json column type put to use! Since this will be in the path of a regularly polled UI route, it may make sense to index `.type` as a virtual column. The json type is stored as binary and avoids serialization overhead but is always going to be costlier to scan than a btree lookup. An example on how to do this: https://dev.mysql.com/doc/refman/5.7/en/create-table-secondary-indexes.html#json-column-indirect-index

Reply:
Awesome, I didn't know about that! Thanks for letting me know!